### ISSUE TITLE
[Bug]:  aws_api_gateway_deployment throwing "all of `stage_description,stage_name` must be specified" error

### ISSUE THREAD

**Issue by grammeaway** [hashicorp/terraform-provider-aws#40073]
### Terraform Core Version

1.8.0

### AWS Provider Version

5.75.0

### Affected Resource(s)

aws_api_gateway_deployment

### Expected Behavior

While running an apply step for a aws_api_gateway_deployment, the deploy should pass, even with the missing stage_name field, since the stage is managed as a separate aws_api_gateway_stage resource (as recommended in the docs as well).

### Actual Behavior

The deployment fails with the error:

```bash
Error: Missing required argument

"stage_description": all of `stage_description,stage_name` must be specified
```

### Relevant Error/Panic Output Snippet

While running Terraform apply with the new provider, we have been getting the 

```bash
"stage_description": all of `stage_description,stage_name` must be specified
```

error, despite running the apply on resources that have been unchanged for a long time. Our setup, rather than using the stage_name field, uses a separate aws_api_gateway_stage reosurce instead, as heavily recommended in the [docs](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/api_gateway_deployment.html ).

When providing a stage_name field, the error disappears, but we end up with the potential issues that using the stage_name field can cause. 

When forcing the TF provider version back to 5.74.0, things also start working again.

### Terraform Configuration Files

```hcl
resource "aws_api_gateway_rest_api" "foo_API" {
  name        = "foo_API"
  description = "Gateway for the foo server"
  disable_execute_api_endpoint = true 
}

resource "aws_api_gateway_deployment" "deployment" {
   depends_on = [
     aws_api_gateway_resource.proxy,
      aws_api_gateway_integration.integration,
      aws_api_gateway_method.method
     
   ]

   stage_description = "${md5(file("api-gateway.tf"))}" 
   rest_api_id = aws_api_gateway_rest_api.foo_API.id

   triggers = {
    redeployment = sha1(jsonencode([
      aws_api_gateway_resource.proxy,
      aws_api_gateway_integration.integration,
      aws_api_gateway_method.method
    ]))
  }

  lifecycle {
    create_before_destroy = true
  }
}

resource "aws_api_gateway_stage" "stage" {
  deployment_id = aws_api_gateway_deployment.deployment.id
  rest_api_id   = aws_api_gateway_rest_api.foo_API.id
  stage_name    = var.stage
}
```

### Steps to Reproduce

Run `terraform apply` with v5.75.0 on an existing aws_api_gateway_rest_api, aws_api_gateway_deployment, aws_api_gateway_stage, where the aws_api_gateway_deployment doesn't have the optional, and potentially error-causing stage_name field. 

### Would you like to implement a fix?

No

**Comment by saleco** [comment]
Im interested in working on this issue.

**Comment by justinretzolk** [comment]
Hey @grammeaway 👋 Thank you for taking the time to raise this! We caught this as well, and have merged a change to fix it (see #40067). We're currently in the process of releasing `5.75.1`, which will include that fix. With that in mind, I'm going to close this issue, but if you experience any further unexpected behavior, please do let us know!
